nextcord/http.py

```diff
@@ -235,151 +708,266 @@ async def request(
+                                else:
+                                    self._dispatch(
+                                        "global_http_ratelimit", global_rate_limit.reset_after
+                                    )
+
+                                if not response.headers.get("Via") or isinstance(ret, str):
+                                    _log.error(
+                                        "Path %s resulted in what appears to be a CloudFlare ban, either a "
+                                        "large amount of errors recently happened and/or Nextcord has a bug."
+                                    )
+                                    # Banned by Cloudflare more than likely.
+                                    raise HTTPException(response, ret)
+
+                                if retry_request:
+                                    _log.warning(
+                                        "Path %s resulted in error 429, rate limit exceeded. Retrying.",
+                                        rate_limit_path,
+                                    )
+                                    should_retry = True
+                                else:
+                                    _log.warning(
+                                        "Path %s resulted in error 429, rate limit exceeded.",
+                                        rate_limit_path,
+                                    )
+                                    raise HTTPException(
+                                        response, ret
+                                    )  # TODO: Make actual HTTPRateLimit error?
+                            elif response.status >= 500:
+                                raise DiscordServerError(response, ret)
+                            else:
+                                raise HTTPException(response, ret)
+
+            # This is handling exceptions from the request
+            except OSError as e:
+                # Connection reset by peer
+                if retry_count < max_retry_count - 1 and e.errno in (54, 10054):
+                    await asyncio.sleep(1 + retry_count * 2)
+                    continue
+
+                raise
+
+            except RateLimitMigrating as e:
+                if url_rate_limit.migrating is None:
+                    raise ValueError(
+                        "RateLimitMigrating raised, but RateLimit.migrating is None. This is an internal Nextcord "
+                        "error and should be reported!"
+                    ) from e
+
+                url_rate_limit = self._buckets.get(url_rate_limit.migrating)
+                if url_rate_limit is None:
+                    # This means we have an internal issue that we need to fix.
+                    raise ValueError(
+                        "RateLimit said to migrate, but the RateLimit to migrate was not found? This is an "
+                        "internal Nextcord error and should be reported!"
+                    ) from e
```

Comment:
Plausibly we also want to do a `_log.error` within this statement with as much context as the library has regarding the situation to help us debug? Cos let's be real, it'll happen sooner or later